pkg/tfvars/azure/azure.go

```diff
@@ -88,6 +90,11 @@ func TFVars(sources TFVarsSources) ([]byte, error) {
 		masterDiskEncryptionSetID = masterConfig.OSDisk.ManagedDisk.DiskEncryptionSet.ID
 	}
 
+	var masterConfigHyperVGeneration string
+	if strings.HasSuffix(masterConfig.Image.ResourceID, "_v2") {
```

Comment:
I am not keen on parsing the image ID to determine the hyperV generation. It is fragile, prone to breaking if a change is made to how we name the image IDs. This will also break if the user uses marketplace images instead, where the image is not specified via an image ID.

Reply:
I tried to bypass the image check. Not sure if the current solution is ideal.